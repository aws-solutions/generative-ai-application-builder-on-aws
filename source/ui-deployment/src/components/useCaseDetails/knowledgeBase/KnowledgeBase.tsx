// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0

import { InfoLink } from '@/components/commons';
import { Container, Header } from '@cloudscape-design/components';
import { KnowledgeBaseDetails } from './KnowledgeBaseDetails';
import { BaseDetailsContainerProps } from '../types';
import { ErrorBoundary } from '@/components/commons/ErrorBoundary';

/**
 * Component that renders a knowledge base container with a header and details
 * @param {Object} props - Component properties
 * @param {Function} props.loadHelpPanelContent - Function to load help panel content
 * @returns {JSX.Element} Knowledge base container with header and details
 */
export const KnowledgeBase = ({
    loadHelpPanelContent,
    selectedDeployment,
    runtimeConfig
}: BaseDetailsContainerProps) => (
    <Container
        header={
            <Header
                variant="h2"
                info={
                    <InfoLink
                        onFollow={() => loadHelpPanelContent(1)}
                        ariaLabel={'Information about deployment knowledge base.'}
                    />
                }
            >
                Knowledge base
            </Header>
        }
        data-testid="knowledge-base-details-container"
    >
        <ErrorBoundary componentName="Knowledge Base Details">
            <KnowledgeBaseDetails selectedDeployment={selectedDeployment} runtimeConfig={runtimeConfig} />
        </ErrorBoundary>
    </Container>
);
