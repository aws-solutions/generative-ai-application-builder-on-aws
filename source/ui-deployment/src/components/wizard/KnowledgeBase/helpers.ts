/**
 * Utility function to convert a numeric relevency score to a Kendra score
 * @param {number} score
 * @returns {string} Kendra score
 */
export const scoreToKendraMapping = (score: number) => {
    if (score < 0 || score > 1.0) {
        throw new Error('Score expected to be between 0 and 1');
    }

    if (score == 1.0) {
        return 'VERY_HIGH';
    } else if (score >= 0.75) {
        return 'HIGH';
    } else if (score >= 0.5) {
        return 'MEDIUM';
    } else if (score >= 0.25) {
        return 'LOW';
    } else {
        //Kendra in the background refers to this state as NOT_AVAILABLE; however, in the context
        //of setting a threshold, DISABLED reads better for customers so using this nomenclature instead
        return 'DISABLED';
    }
};
