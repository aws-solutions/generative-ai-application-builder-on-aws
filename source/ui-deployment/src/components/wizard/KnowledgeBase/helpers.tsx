import { IG_DOCS } from '@/utils/constants';
import { Box, SpaceBetween } from '@cloudscape-design/components';

/**
 * Utility function to convert a numeric relevance score to a Kendra score
 * @param {number} score
 * @returns {string} Kendra score
 */
export const scoreToKendraMapping = (score: number) => {
    if (score < 0 || score > 1.0) {
        throw new Error('Score expected to be between 0 and 1');
    }

    if (score == 1.0) {
        return 'VERY_HIGH';
    } else if (score >= 0.75) {
        return 'HIGH';
    } else if (score >= 0.5) {
        return 'MEDIUM';
    } else if (score >= 0.25) {
        return 'LOW';
    } else {
        //Kendra in the background refers to this state as NOT_AVAILABLE; however, in the context
        //of setting a threshold, DISABLED reads better for customers so using this nomenclature instead
        return 'DISABLED';
    }
};

export const knowledgeBaseInfoPanel = {
    default: {
        title: 'Knowledge base selection',
        content: (
            <div>
                <Box variant="p">
                    Use this page to configure a knowledge base to enable <b>Retrieval Augmented Generation (RAG)</b>.
                    Connecting a knowledge base to the deployment enables it to source additional information to pass
                    onto the LLM. Disabling RAG uses the LLM as is.
                </Box>

                <Box variant="p">
                    RAG is a technique that leverages prompt-based learning to “train” a LLM on new information it
                    wasn’t originally trained on. Instead of using the traditional method of fine-tuning a model on
                    custom data, prompt-based learning and RAG rely on a concept called{' '}
                    <b>
                        <i>context stuffing</i>
                    </b>
                    . By “stuffing” additional “context” into your prompt, you can show the LLM new information it
                    hasn’t yet seen such as private documents, technical jargon, the latest news articles, etc.
                </Box>
            </div>
        ),
        links: [
            {
                href: IG_DOCS.INGESTING_DATA,
                text: 'Ingesting data into your Knowledge base'
            }
        ]
    },
    kendraIndex: {
        title: 'Amazon Kendra index',
        content: (
            <SpaceBetween size="xs">
                <Box variant="p">
                    This solution can use an existing Kendra index as the knowledge base, or it can create a new one for
                    you.
                </Box>

                <Box variant="p">
                    If you choose to use an existing index, you will need to provide the index id, which can be found on
                    the Amazon Kendra console. Please ensure the Amazon Kendra index is active and populated with your
                    desired data.
                </Box>

                <Box variant="p">
                    If creating a new Amazon Kendra index for this application, please provide a name for the index.
                </Box>

                <Box variant="p">
                    <b>Note:</b> You may need to contact your AWS Account Administrator for help with certain actions
                    such as ingesting documents into the index or deleting resources once they are no longer needed.
                </Box>
            </SpaceBetween>
        ),
        links: [
            {
                href: IG_DOCS.INGESTING_DATA,
                text: 'Ingesting data into your Knowledge base'
            },
            {
                href: IG_DOCS.CONCEPTS,
                text: 'Concepts and Definitions - DevOps user'
            },
            {
                href: IG_DOCS.CLEANUP_KENDRA,
                text: 'Cleaning up Kendra indexes'
            }
        ]
    },
    kendraAdditionalQueryCapacity: {
        title: 'Additional Amazon Kendra options',
        content: (
            <SpaceBetween size="xs">
                <Box variant="h4">Kendra additional query capacity</Box>
                <Box variant="p">
                    Amazon Kendra provides resources for your index in capacity units. Each capacity unit provides
                    additional resources for your index to use. Use this setting to adjust the query capacity for your
                    index.
                </Box>

                <Box variant="h4">Kendra additional storage capacity</Box>
                <Box variant="p">Use this setting to adjust the storage capacity for your index.</Box>

                <Box variant="h4">Kendra edition</Box>
                <Box variant="p">
                    Amazon Kendra comes in two editions. Kendra Enterprise Edition provides a high-availability service
                    for production workloads. Kendra Developer Edition provides developers with a lower-cost option to
                    build a proof-of-concept. Use this setting to select the type of Kendra index to create for this
                    deployment.
                </Box>
            </SpaceBetween>
        ),
        links: [
            {
                href: 'https://docs.aws.amazon.com/kendra/latest/dg/adjusting-capacity.html',
                text: 'Amazon Kendra - Adjusting Capacity'
            }
        ]
    }
};
