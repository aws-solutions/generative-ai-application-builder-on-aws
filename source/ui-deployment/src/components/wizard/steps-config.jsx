/**********************************************************************************************************************
 *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.                                                *
 *                                                                                                                    *
 *  Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance    *
 *  with the License. A copy of the License is located at                                                             *
 *                                                                                                                    *
 *      http://www.apache.org/licenses/LICENSE-2.0                                                                    *
 *                                                                                                                    *
 *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES *
 *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions    *
 *  and limitations under the License.                                                                                *
 *********************************************************************************************************************/
import {
    DEFAULT_KENDRA_NUMBER_OF_DOCS,
    DEFAULT_ADDITIONAL_KENDRA_QUERY_CAPACITY,
    DEFAULT_ADDITIONAL_KENDRA_STORAGE_CAPACITY
} from '../../utils/constants';

export const USE_CASE_OPTIONS = [
    {
        value: 'Chat',
        label: 'Chat'
    }
];

export const KENDRA_EDITIONS = [
    { value: 'developer', label: 'Developer' },
    { value: 'enterprise', label: 'Enterprise' }
];

export const KNOWLEDGE_BASE_TYPES = [{ value: 'kendra', label: 'Kendra' }];

export const BEDROCK_MODEL_OPTION_IDX = 0;
export const ANTHROPIC_MODEL_OPTION_IDX = 1;
export const HF_MODEL_OPTION_IDX = 2;
export const HF_INF_ENDPOINT_OPTION_IDX = 3;

export const MODEL_FAMILY_PROVIDER_OPTIONS = [
    {
        label: 'Bedrock',
        value: 'Bedrock'
    },
    {
        label: 'Anthropic',
        value: 'Anthropic'
    },
    {
        label: 'HuggingFace',
        value: 'HuggingFace'
    },
    {
        label: 'HuggingFace - Inference Endpoint',
        value: 'HuggingFace-InferenceEndpoint'
    }
];

export const INCLUDE_UI_OPTIONS = [
    {
        value: 'yes',
        label: 'Yes',
        description: 'A UI will be created and deployed with the backend API'
    },
    {
        value: 'no',
        label: 'No',
        description: 'A backend API will be created and deployed. You may connect your own UI to the backend API'
    }
];

export const DEFAULT_STEP_INFO = {
    useCase: {
        useCase: USE_CASE_OPTIONS[0],
        useCaseName: '',
        useCaseDescription: '',
        defaultUserEmail: '',
        inError: false
    },
    knowledgeBase: {
        isRagRequired: false,
        knowledgeBaseType: KNOWLEDGE_BASE_TYPES[0],
        existingKendraIndex: '',
        kendraIndexId: '',
        kendraAdditionalQueryCapacity: DEFAULT_ADDITIONAL_KENDRA_QUERY_CAPACITY,
        kendraAdditionalStorageCapacity: DEFAULT_ADDITIONAL_KENDRA_STORAGE_CAPACITY,
        kendraEdition: KENDRA_EDITIONS[0],
        maxNumDocs: DEFAULT_KENDRA_NUMBER_OF_DOCS,
        inError: false,
        kendraIndexName: ''
    },
    model: {
        modelProvider: MODEL_FAMILY_PROVIDER_OPTIONS[0],
        apiKey: '',
        modelName: '',
        modelFamily: '',
        promptTemplate: '',
        inferenceEndpoint: '',
        modelParameters: [],
        inError: false,
        temperature: 0.1,
        verbose: false,
        streaming: false
    }
};

export const BEDROCK_MODEL_PROVIDERS_WITH_STREAMING = ['amazon', 'anthropic'];
