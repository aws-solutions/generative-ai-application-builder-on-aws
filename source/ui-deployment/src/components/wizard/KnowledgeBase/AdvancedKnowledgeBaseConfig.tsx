/**********************************************************************************************************************
 *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.                                                *
 *                                                                                                                    *
 *  Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance    *
 *  with the License. A copy of the License is located at                                                             *
 *                                                                                                                    *
 *      http://www.apache.org/licenses/LICENSE-2.0                                                                    *
 *                                                                                                                    *
 *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES *
 *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions    *
 *  and limitations under the License.                                                                                *
 **********************************************************************************************************************/

import { AdvancedKnowledgeBaseConfigProps } from '../interfaces/Steps';
import { Container, Header, SpaceBetween } from '@cloudscape-design/components';
import { RetrieveDocumentCount } from './common/RetrieveDocumentCount';
import { ReturnSourceDocuments } from './common/ReturnSourceDocuments';
import { EnableRoleBasedAccessControl } from './common/EnableRoleBasedAccessControl';
import { KNOWLEDGE_BASE_PROVIDERS } from '../steps-config';
import { AttributeFilterEditor } from './providers/Kendra/AttributeFilterEditor';
import RetrievalFilterEditor from './providers/Bedrock/RetrievalFilterEditor';
import OverrideSearchType from './providers/Bedrock/OverrideSearchType';
import ScoreThreshold from './common/ScoreThreshold';
import NoDocsFoundResponse from './common/NoDocsFoundResponse';

export const AdvancedKnowledgeBaseConfig = (props: AdvancedKnowledgeBaseConfigProps) => {
    return (
        <Container
            header={<Header variant="h2">Advanced RAG configurations</Header>}
            data-testid="advanced-knowledgebase-config-container"
        >
            <SpaceBetween size="l">
                <RetrieveDocumentCount {...props} />
                <ScoreThreshold
                    scoreThreshold={props.knowledgeBaseData.scoreThreshold}
                    knowledgeBaseProvider={props.knowledgeBaseData.knowledgeBaseType.value}
                    data-testid="knowledge-base-score-threshold"
                    {...props}
                />
                <NoDocsFoundResponse
                    noDocsFoundResponse={props.knowledgeBaseData.noDocsFoundResponse}
                    data-testid="knowledge-base-no-docs-found-response"
                    {...props}
                />
                <ReturnSourceDocuments {...props} />
                {props.knowledgeBaseData.knowledgeBaseType.value === KNOWLEDGE_BASE_PROVIDERS.kendra && (
                    <>
                        <EnableRoleBasedAccessControl {...props} />
                        <AttributeFilterEditor {...props} />
                    </>
                )}
                {props.knowledgeBaseData.knowledgeBaseType.value === KNOWLEDGE_BASE_PROVIDERS.bedrock && (
                    <>
                        <RetrievalFilterEditor {...props} />
                        <OverrideSearchType {...props} />
                    </>
                )}
            </SpaceBetween>
        </Container>
    );
};

export default AdvancedKnowledgeBaseConfig;
