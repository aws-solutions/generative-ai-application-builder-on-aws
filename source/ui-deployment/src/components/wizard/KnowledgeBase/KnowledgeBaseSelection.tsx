/**********************************************************************************************************************
 *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.                                                *
 *                                                                                                                    *
 *  Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance    *
 *  with the License. A copy of the License is located at                                                             *
 *                                                                                                                    *
 *      http://www.apache.org/licenses/LICENSE-2.0                                                                    *
 *                                                                                                                    *
 *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES *
 *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions    *
 *  and limitations under the License.                                                                                *
 **********************************************************************************************************************/

import React from 'react';
import { KnowledgeBaseConfigProps } from '../interfaces/Steps';
import { KNOWLEDGE_BASE_PROVIDERS } from '../steps-config';
import { Kendra } from './providers';
import { BedrockKnowledgeBase } from './providers/Bedrock';

export const KnowledgeBaseSelection = (props: KnowledgeBaseConfigProps) => {
    React.useEffect(() => {
        if (props.knowledgeBaseData.knowledgeBaseType.value === KNOWLEDGE_BASE_PROVIDERS.bedrock) {
            props.setRequiredFields!(['bedrockKnowledgeBaseId']);
        }
        // reset the error
        props.onChangeFn({ inError: false });
    }, [props.knowledgeBaseData.knowledgeBaseType]);

    if (props.knowledgeBaseData.knowledgeBaseType.value === KNOWLEDGE_BASE_PROVIDERS.kendra) {
        return <Kendra {...props} />;
    } else if (props.knowledgeBaseData.knowledgeBaseType.value === KNOWLEDGE_BASE_PROVIDERS.bedrock) {
        return <BedrockKnowledgeBase {...props} />;
    }
    return <div>Invalid knowledgebase type</div>;
};

export default KnowledgeBaseSelection;
