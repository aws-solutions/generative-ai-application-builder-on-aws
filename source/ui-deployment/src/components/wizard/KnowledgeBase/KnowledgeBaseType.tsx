/**********************************************************************************************************************
 *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.                                                *
 *                                                                                                                    *
 *  Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance    *
 *  with the License. A copy of the License is located at                                                             *
 *                                                                                                                    *
 *      http://www.apache.org/licenses/LICENSE-2.0                                                                    *
 *                                                                                                                    *
 *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES *
 *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions    *
 *  and limitations under the License.                                                                                *
 **********************************************************************************************************************/

import { Container, FormField, Header, Select, SelectProps } from '@cloudscape-design/components';
import { BaseFormComponentProps } from '../interfaces';
import { InfoLink } from '../../commons';
import { KNOWLEDGE_BASE_TYPES } from '../steps-config';
import { TOOLS_CONTENT } from '../tools-content';

const { knowledgeBase: knowledgeBaseToolsContent } = TOOLS_CONTENT;

interface KnowledgeBaseTypeProps extends BaseFormComponentProps {
    knowledgeBaseData: any;
}

export const KnowledgeBaseType = (props: KnowledgeBaseTypeProps) => {
    const onKnowledgeBaseTypeChange = (detail: SelectProps.ChangeDetail) => {
        props.onChangeFn({
            knowledgeBaseType: detail.selectedOption
        });
    };

    return (
        <Container
            header={<Header variant="h2">Select Knowledge Base</Header>}
            data-testid="select-knowledgebase-type-container"
        >
            <FormField
                label="Knowledge base type"
                info={
                    <InfoLink
                        onFollow={() => props.setHelpPanelContent!(knowledgeBaseToolsContent.default)}
                        ariaLabel={'Information about different knowledge bases.'}
                    />
                }
                description="Select a supported knowledge base to attach to the deployment."
            >
                <Select
                    options={KNOWLEDGE_BASE_TYPES}
                    onChange={({ detail }) => onKnowledgeBaseTypeChange(detail)}
                    selectedAriaLabel="Selected"
                    selectedOption={props.knowledgeBaseData.knowledgeBaseType}
                />
            </FormField>
        </Container>
    );
};

export default KnowledgeBaseType;
