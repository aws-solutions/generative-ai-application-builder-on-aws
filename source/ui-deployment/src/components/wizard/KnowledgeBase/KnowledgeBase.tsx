/**********************************************************************************************************************
 *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.                                                *
 *                                                                                                                    *
 *  Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance    *
 *  with the License. A copy of the License is located at                                                             *
 *                                                                                                                    *
 *      http://www.apache.org/licenses/LICENSE-2.0                                                                    *
 *                                                                                                                    *
 *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES *
 *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions    *
 *  and limitations under the License.                                                                                *
 *********************************************************************************************************************/

import React from 'react';
import {
    Box,
    Container,
    Header,
    FormField,
    SpaceBetween,
    RadioGroup,
    RadioGroupProps
} from '@cloudscape-design/components';
import { InfoLink } from '../../commons/common-components';
import { TOOLS_CONTENT } from '../tools-content.jsx';

import AdvancedKnowledgeBaseConfig from './AdvancedKnowledgeBaseConfig';
import { KnowledgeBaseType } from './KnowledgeBaseType';
import KnowledgeBaseSelection from './KnowledgeBaseSelection';
import { StepContentProps } from '../interfaces/Steps';

const { knowledgeBase: knowledgeBaseToolsContent } = TOOLS_CONTENT;

export const KnowledgeBase = ({ info: { knowledgeBase }, setHelpPanelContent, onChange }: StepContentProps) => {
    const [numFieldsInError, setNumFieldsInError] = React.useState(0);

    const initRagSelectedOption = () => {
        if (knowledgeBase.isRagRequired) {
            return 'true';
        }
        return 'false';
    };

    const [ragSelectedOption, setRagSelectedOption] = React.useState(initRagSelectedOption);
    const [requiredFields, setRequiredFields] = React.useState<any>([]);

    const isRequiredFieldsFilled = () => {
        for (const field of requiredFields) {
            if (knowledgeBase[field].length === 0) {
                return false;
            }
        }
        return true;
    };

    const updateError = () => {
        if (numFieldsInError > 0 || !isRequiredFieldsFilled()) {
            onChange({ inError: true });
        } else if (numFieldsInError === 0 && isRequiredFieldsFilled()) {
            onChange({ inError: false });
        }
    };

    React.useEffect(() => {
        updateError();
        // eslint-disable-next-line react-hooks/exhaustive-deps
    }, [
        numFieldsInError,
        requiredFields,
        knowledgeBase.kendraIndexName,
        knowledgeBase.kendraIndexId,
        knowledgeBase.kendraAdditionalQueryCapacity,
        knowledgeBase.kendraAdditionalStorageCapacity,
        knowledgeBase.maxNumDocs,
        knowledgeBase.knowledgeBaseType,
        knowledgeBase.bedrockKnowledgeBaseId,
        knowledgeBase.bedrockOverrideSearchType
    ]);

    const handleRagSelection = (detail: RadioGroupProps.ChangeDetail) => {
        setRagSelectedOption(detail.value);
        setRequiredFields([]);
        knowledgeBase.isRagRequired = detail.value === 'true';

        // set error state to false if rag is off
        if (detail.value === 'false') {
            setNumFieldsInError(0);
            onChange({ inError: false });
        }
    };

    return (
        <Box margin={{ bottom: 'l' }} data-testid="rag-required-dropdown">
            <SpaceBetween size="l">
                <Container
                    header={<Header variant="h2">Retrieval Augmented Generation (RAG)</Header>}
                    data-testid="rag-required-container"
                >
                    <SpaceBetween size="l">
                        <FormField
                            label="Do you want to enable Retrieval Augmented Generation (RAG) for this use case?"
                            description="If RAG is enabled, a Knowledge Base (such as one powered by Amazon Kendra) is required to be configured."
                            info={
                                <InfoLink
                                    onFollow={() => setHelpPanelContent(knowledgeBaseToolsContent.default)}
                                    ariaLabel={'Information about enabling RAG.'}
                                />
                            }
                        >
                            <RadioGroup
                                onChange={({ detail }) => handleRagSelection(detail)}
                                items={[
                                    {
                                        value: 'true',
                                        label: 'Yes'
                                    },
                                    {
                                        value: 'false',
                                        label: 'No'
                                    }
                                ]}
                                value={ragSelectedOption}
                            />
                        </FormField>
                    </SpaceBetween>
                </Container>

                {knowledgeBase.isRagRequired && (
                    <KnowledgeBaseType
                        onChangeFn={onChange}
                        setHelpPanelContent={setHelpPanelContent}
                        knowledgeBaseData={knowledgeBase}
                        setNumFieldsInError={setNumFieldsInError}
                    />
                )}

                {knowledgeBase.isRagRequired && (
                    <KnowledgeBaseSelection
                        onChangeFn={onChange}
                        knowledgeBaseData={knowledgeBase}
                        setHelpPanelContent={setHelpPanelContent}
                        setNumFieldsInError={setNumFieldsInError}
                        setRequiredFields={setRequiredFields}
                    />
                )}

                {knowledgeBase.isRagRequired && (
                    <AdvancedKnowledgeBaseConfig
                        onChangeFn={onChange}
                        setHelpPanelContent={setHelpPanelContent}
                        knowledgeBaseData={knowledgeBase}
                        setNumFieldsInError={setNumFieldsInError}
                    />
                )}
            </SpaceBetween>
        </Box>
    );
};
export default KnowledgeBase;
