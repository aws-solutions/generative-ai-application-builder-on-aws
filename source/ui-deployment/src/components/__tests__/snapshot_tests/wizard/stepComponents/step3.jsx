/**********************************************************************************************************************
 *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.                                                *
 *                                                                                                                    *
 *  Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance    *
 *  with the License. A copy of the License is located at                                                             *
 *                                                                                                                    *
 *      http://www.apache.org/licenses/LICENSE-2.0                                                                    *
 *                                                                                                                    *
 *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES *
 *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions    *
 *  and limitations under the License.                                                                                *
 **********************************************************************************************************************/

import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import HomeContext from '../../../../../home/home.context';
import KnowledgeBase from '../../../../wizard/stepComponents/step3';
import { DEFAULT_STEP_INFO } from '../../../../wizard/steps-config';

jest.mock('@cloudscape-design/components', () => {
    const Components = jest.genMockFromModule('@cloudscape-design/components');
    for (const componentName of Object.keys(Components)) {
        Components[componentName] = componentName;
    }
    return Components;
});

describe('Chat', () => {
    const contextValue = {
        'dispatch': jest.fn(),
        'state': {
            'selectedDeployment': {},
            'deploymentsData': [],
            'deploymentAction': 'CREATE',
            'authorized': true
        }
    };

    test('Snapshot test for wizard step 3', async () => {
        const tree = renderer
            .create(
                <>
                    <HomeContext.Provider
                        value={{
                            ...contextValue
                        }}
                    >
                        <KnowledgeBase info={DEFAULT_STEP_INFO} onChange={jest.fn()} setHelpPanelContent={jest.fn()} />
                    </HomeContext.Provider>
                </>
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});
