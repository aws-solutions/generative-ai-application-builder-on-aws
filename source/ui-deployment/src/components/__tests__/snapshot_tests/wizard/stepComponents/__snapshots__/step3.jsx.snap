// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chat Snapshot test for wizard step 3 1`] = `
<Box
  data-testid="rag-required-dropdown"
  margin={
    Object {
      "bottom": "l",
    }
  }
>
  <SpaceBetween
    size="l"
  >
    <Container
      data-testid="rag-required-container"
      header={
        <Header
          variant="h2"
        >
          Retrieval Augmented Generation (RAG)
        </Header>
      }
    >
      <SpaceBetween
        size="l"
      >
        <FormField
          description="If RAG is enabled, a Knowledge Base (such as one powered by Amazon Kendra) is required to be configured."
          info={
            <InfoLink
              ariaLabel="Information about enabling RAG."
              onFollow={[Function]}
            />
          }
          label="Do you want to enable Retrieval Augmented Generation (RAG) for this use case?"
        >
          <Select
            onChange={[Function]}
            options={
              Array [
                Object {
                  "label": "yes",
                  "value": true,
                },
                Object {
                  "label": "no",
                  "value": false,
                },
              ]
            }
            selectedOption={
              Object {
                "label": "no",
                "value": false,
              }
            }
          />
        </FormField>
      </SpaceBetween>
    </Container>
  </SpaceBetween>
</Box>
`;
